src/Message/Refund.php

```diff
@@ -0,0 +1,377 @@
+     *
+     * @return null|string
+     * @deprecated see getRefundReference
+     */
+    public function getReference()
+    {
+        return $this->getRefundReference();
+    }
+
+    /**
+     * Set the refund reference
+     *
+     * @param string $value
+     * @return static
+     * @deprecated see setRefundReference
+     */
+    public function setReference($value)
+    {
+        return $this->setRefundReference($value);
+    }
+
+    /**
+     * Get the currency
+     *
+     * @return null|string
+     */
+    public function getCurrency()
+    {
+        return $this->getParameter('currency');
+    }
+
+    /**
+     * Set the currency
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setCurrency($value)
+    {
+        return $this->setParameter('currency', $value);
+    }
+
+    /**
+     * Get the monetary amount
+     *
+     * @return null|string
+     */
+    public function getAmount()
+    {
+        return $this->getParameter('amount');
+    }
+
+    /**
+     * Set the monetary amount
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setAmount($value)
+    {
+        return $this->setParameter('amount', $value);
+    }
+
+    /**
+     * Get the reason
+     *
+     * @return null|string
+     */
+    public function getReason()
+    {
+        return $this->getParameter('reason');
+    }
+
+    /**
+     * Set the reason
+     *
+     * @param string $value
+     * @return static
+     */
+    public function setReason($value)
+    {
+        return $this->setParameter('reason', $value);
+    }
+
+    /**
+     * Get the reason
+     *
+     * @return null|string
+     * @deprecated in favor of getReason
+     */
+    public function getNote()
+    {
+        return $this->getReason();
+    }
+
+    /**
+     * Set the reason
+     *
+     * @param string $value
+     * @return static
+     * @deprecated in favor of setReason
+     */
+    public function setNote($value)
+    {
+        return $this->setReason($value);
+    }
+
+    /**
+     * Get the time
+     *
+     * @return DateTime|null
+     */
+    public function getTime()
+    {
+        return $this->getParameter('time');
+    }
+
+    /**
+     * Set the time
+     *
+     * @param DateTime|null $value
+     * @return static
+     */
+    public function setTime($value)
+    {
+        return $this->setParameter('time', $value);
+    }
+
+    /**
+     * Get the transaction
+     *
+     * @return null|Transaction
+     */
+    public function getTransaction()
+    {
+        return $this->getParameter('transaction');
+    }
+
+    /**
+     * Set the transaction
+     *
+     * @param Transaction $value
+     * @return static
+     */
+    public function setTransaction($value)
+    {
+        return $this->setParameter('transaction', $value);
+    }
+
+    /**
+     * Get the transaction id
+     *
+     * @return null|string
+     */
+    public function getTransactionId()
```

Comment:
There's a lot of overlap between the `Refund` and `Chargeback` classes. Maybe we need a parent class `Incident` that these two can extend?

Reply:
These two classes were basically pulled from the `omnipay-repo`. I'd like to do some work to consolidate and standardize these in a separate package that can be shared but it's really out of scope for now. I do like your suggestions but I would prefer to have that be tackled in conjunction with `omnipay-vindicia` at a later point.

Reply:
Yeah that sounds good. We could do a big refactor of omnipay at some point and add a package containing all the different entities that could be sent or received from a payment gateway.